### PR TITLE
Flow and helpers

### PR DESCRIPTION
**Pull request by davidmiller**
This is most of a train journey to NCL's worth of things rolled up into one.

You'll need to pull Odontotheme master again, and run the migrations.

I would actually quite like to put something like `pathway.as_menuitem` `pathway.get_absolute_url` and the `link_to` `button_to` templatetags into Opal core, but here for now to see how the implementation feels.

Bunch of bits and pieces here: 

* change/delete some fields
* add defaults for the pathway (should de hardcode these later)
* update the detail page layout and display (note banner and demographics edit)
* take the fp17 form out of the menu - it only makes sense in the context of a patient
* lots of cosmetic improvements to forms and record display

Later on, I'm fairly sure we need to look at the models, I think we'll get a lot of mileage from refactoring. Will figure out how once I've done some more clinic visits.
We're also going to need Episode category stages I'm sure, and almost certainly different FP17 forms (for start treatment and complete treatment) - but let's see !

### CODE REVIEW

**Comment by pacharanero** [comment]
> I would actually quite like to put something like pathway.as_menuitem pathway.get_absolute_url and the link_to button_to templatetags into Opal core, but here for now to see how the implementation feels.
These sound like a good idea.

**Comment by pacharanero** [comment]
All the input boxes with `maxlength="2"` are registering errors although there does not appear to be any content in them

image

**Comment by pacharanero** [comment]
The pathway button is broken

image

I'm not sure what causes this but it seems to happen when there is an error further up in the Pathway.

**Comment by pacharanero** [comment]
I do like the improved workflow around adding a patient first and then creating an episode

**Comment by davidmiller** [comment]
> `The pathway button is broken` 

@pacharanero That's interesting - doesn't happen to me!

Normally there will be a console error if it's doing that...

**Comment by davidmiller** [comment]
> All the input boxes with maxlength="2" are registering errors although there does not appear to be any content in them

This is probably the same thing as the error in the pathway button - e.g. the Angular that drives that bit of markup is erroring out and so it doesn't run far enough through to `ng-hide` the error messages.

**Comment by davidmiller** [comment]
> I do like the improved workflow around adding a patient first and then creating an episode

It doesn't _quite_ do that :)
(I tried this first actually, but it turned out to be quite involved. One of the enabling steps would be the refactor discussed in https://github.com/openhealthcare/opal/issues/1433)

Rather, we're making significant changes to the episode detail layout based on key episode data like 'is there a date of acceptance / completion', which turned out to be much easier to work with, if slightly less clean.

What happens as you transition across episodes will require some thought, but stages first almost certainly helps us think about it....

File: odonto/templates/forms/demographics_form.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{% load forms %}
+{% input field="Demographics.nhs_number" label="NHS Number" %}
```
Comment (fredkingham) [inline odonto/templates/forms/demographics_form.html]: FYI you shouldn't need 'NHS Number' its the verbose name of the field (I'm sure this is cut and pasted from opal but we should change it there too)

File: odonto/settings.py [hunk 2]
```diff
@@ -281,6 +281,8 @@
     )
 }
 
+ODONTO_LOGIN_MESSAGE = 'Log in to get started'
```
Comment (fredkingham) [inline odonto/settings.py]: Is this used anywhere?

**Comment by fredkingham** [comment]
@davidmiller looks great, no deal breakers, do we actually use that setting? You might want to remove it. I'll put in a pull request to remove the label from the opal demographics form nhs number field now.

File: odonto/settings.py (hunk 2 above)
Comment (davidmiller) [reply odonto/settings.py]: Yes. It's the log in message.

**Comment by davidmiller** [comment]
https://github.com/odonto/opal-odontotheme/blob/dbade54a4c208157ade11e6e628776ebe6405a7a/odontotheme/templates/registration/login.html

**Pull request merged**